src/Platform/Microsoft.Testing.Extensions.HangDump/HangDumpProcessLifetimeHandler.cs

```diff
@@ -222,7 +222,7 @@ public async Task OnTestHostProcessExitedAsync(ITestHostProcessInformation testH
 
     private async Task ActivityTimerAsync()
```

Comment:
This method should be sync, Mutex works with thread affinity and await is changing the thread.